fn/t2.go

```diff
@@ -0,0 +1,24 @@
+package fn
+
+// T2 is the simplest 2-tuple type. It is useful for capturing ad hoc
+// type conjunctions in a single value that can be easily dotchained.
+type T2[A, B any] struct {
+	fst A
+	snd B
+}
+
+// Fst returns the first value in the T2.
+func (t2 T2[A, B]) Fst() A {
```

Comment:
Yeah, definitely. I only chose this because it's [what I know](https://hackage.haskell.org/package/base-4.20.0.1/docs/Prelude.html#v:fst). Will rename.